### ISSUE TITLE
I found a tar.gz that broke ratarmount after 16 hours ;-;

### ISSUE THREAD

**Issue by mr-bo-jangles** [mxmlnkn/ratarmount#66]
Currently at position 1664643022848 of 1664666965947 (100.00%). Estimated time remaining with current rate: 0 min 1 s, with average rate: 0 min 0 s.
Creating offset dictionary for /home/download/thefile.tar.gz took 57789.62s

Traceback (most recent call last):
  File "/usr/local/bin/ratarmount", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/dist-packages/ratarmount.py", line 2604, in cli
    fuseOperationsObject = TarMount(
  File "/usr/local/lib/python3.8/dist-packages/ratarmount.py", line 1937, in __init__
    self.mountSources: List[Union[SQLiteIndexedTar, FolderMountSource]] = [
  File "/usr/local/lib/python3.8/dist-packages/ratarmount.py", line 1938, in <listcomp>
    SQLiteIndexedTar(tarFile, writeIndex=True, **sqliteIndexedTarOptions)
  File "/usr/local/lib/python3.8/dist-packages/ratarmount.py", line 504, in __init__
    self._loadOrStoreCompressionOffsets()  # store
  File "/usr/local/lib/python3.8/dist-packages/ratarmount.py", line 1622, in _loadOrStoreCompressionOffsets
    db.execute('INSERT INTO gzipindex VALUES (?)', (file.read(),))
OverflowError: BLOB longer than INT_MAX bytes

**Comment by mxmlnkn** [comment]
Thank you for reporting this! And sorry about your wasted computation power. I was not aware that there is a maximum blob size even though I did some quite large benchmarks. The limit seems to be around [1GB](https://www.sqlite.org/limits.html) for the blob size. I might be able to reproduce this problem by myself by choosing smaller `indexed_gzip` seek points.

Conversely you might be able to work around your issue for now by increasing the gzip seek point distance, which reduces the data to write out but it might slow down random seeking a little bit. One seek point takes up roughly 32kiB. The default spacing should be 16MiB, i.e., the index required for gzip seeking is ~0.2% of your original files. As you hit the 1GB index limit, this means your file must be larger than 512GB. Is that correct? If not, then my assumptions are wrong! If it is correct and if you want to try again, then please try:

    ratarmount --gzip-seek-point-spacing 128 ...

This will increase the gzip spacing to 128MiB and ratarmount should work if your tar.gz is smaller than 1GB/32kiB*128MiB = 4 TiB. If your archive is even larger or close to it then please choose an appropriately higher seek point spacing.

That SQLite database is essential for ratarmount to work even if it isn't written to disk. Therefore, it might take me a bit to fix this. I guess, I'll have to split the data into multiple smaller blobs to avoid this limit. The limit also can't be increased much more, only to the 2GB limit (signed 32 bit max number).

**Comment by mr-bo-jangles** [comment]
> According to your output log, your file seems to be about 155GiB. Is that correct?
The archive itself is 1.51TB, so you're slightly off

> The limit also can't be increased much more, only to the 2GB limit (signed 32 bit max number). I guess, I'll have to split the data into multiple smaller blobs to avoid the the limit.
This is almost certainly the best choice

**Comment by mxmlnkn** [comment]
>> According to your output log, your file seems to be about 155GiB. Is that correct?
>
> The archive itself is 1.51TB, so you're slightly off

I must have forgotten a 0 somewhere. Well, `--gzip-seek-point-spacing 128` might still work but a value of 512 or so might be safer.

Sorry about editing your post. I wanted to quote it not edit it...

**Comment by mxmlnkn** [comment]
I pushed a fix. You can try it out with:

```bash
pip install git+https://github.com/mxmlnkn/ratarmount.git@fix-gzindex-max-blob-size#egginfo=ratarmount
```

**Comment by mr-bo-jangles** [comment]
I'll be without internet for a few weeks, so hopefully I can try this when I get internet again

**Comment by mxmlnkn** [comment]
Hopefully fixed in 0.8.1. Please let me know if it also works for you and if not feel free to reopen this issue.
